crates/forge/src/lib.rs

```diff
@@ -462,35 +488,101 @@ fn run_with_fuzzing(
+                },
+                () = runner_config.exit_first_cancellation_token.cancelled() => {
+                    //Stop executing all tests because flag --exit-first'
+                    //has been set and one test FAIL
+                    Ok(TestCaseSummary::skipped(&case_copy))
+                },
+                () = cancellation_fuzzing_token.cancelled() => {
+                    //Stop executing all single fuzzing tests
+                    //because one of fuzzing test has been FAIL
+                    Ok(TestCaseSummary::skipped(&case_copy))
+                },
+               result = blocking_run_from_test(
+                    runner,
+                    case,
+                    runner_config.clone(),
+                    args.clone(),
+                    Some(send),
+                ) => {
+                    match result {
+                        Ok(result) => {
+                            if let TestCaseSummary::Failed { .. } = &result {
+                                if runner_config.exit_first {
+                                    runner_config.exit_first_cancellation_token.cancel();
+                                } else {
+                                    cancellation_fuzzing_token.cancel();
+                                }
+                            }
+                            Ok(result)
+                        }
+                        Err(e) => {
+                            runner_config.error_cancellation_token.cancel();
+                            Err(e)
+                        }
+                    }
+                },
+            }
+        }));
+    }
 
-        let result = run_from_test_case(
-            package_root,
-            runner,
-            case,
-            runner_config.fork_targets.as_ref(),
-            &runner_params.contracts,
-            &runner_params.predeployed_contracts,
-            args.clone(),
-            &runner_params.environment_variables,
-        )?;
+    let mut results = vec![];
+
+    drop(send);
+
+    while let Some(task) = tasks.next().await {
+        let result = task??;
         results.push(result.clone());
 
-        if let TestCaseSummary::Failed { .. } = result {
-            // Fuzz failed
-            break;
+        match &result {
+            TestCaseSummary::Failed { .. } | TestCaseSummary::None {} => {
+                break;
+            }
+            _ => (),
         }
     }
 
-    let result = results
+    let _ = recv.recv().await;
+
+    let runs = u32::try_from(
+        results
+            .iter()
+            .filter(|item| {
+                matches!(
+                    item,
+                    TestCaseSummary::Passed { .. } | TestCaseSummary::Failed { .. }
+                )
+            })
+            .count(),
+    )?;
+
+    let result: TestCaseSummary = results
         .last()
         .expect("Test should always run at least once")
         .clone();
```

Comment:
This check should probably be executed before counting runs.

Reply:
I'm not sure what do you mean

Reply:
We should first try to get `result` from `results` before we count the runs. So we panic early if no result is present.